### ISSUE TITLE
Maximize Workbench Editor

### ISSUE THREAD

**Issue by benibenj** [microsoft/vscode#196722]
We want to allow users to to not only maximize an editor group, but also the editor itself across the workbench.
I suggest to add the following command which can be added to the double click action of editor tabs:

`Toggle Maximize Editor` (Naming still open)

Double clicking the first time:
- Stores current state: visibility of Side Panels & Auxiliary Bar, ...
- Maximizes Editor Group
- Hides Side Panels & Auxiliary Bar

Double clicking the second time:
- Unmaximize Editor Group
- If the visibility of the panels has been shown in the meantime:
  - Do not restore saved state
 - If all panels are still hidden
   - Restore the visibility of the panels that were initially visible

**Comment by garretwilson** [comment]
As discussed in #16927, one major point is that this should happen wherever the focus is. If we are in the terminal, hitting the accelerator key will maximize the terminal (like `Ctrl+Shift+J` does now in VS Codeon Windows). If you're in the editor, it maximizes the editor (similar but not completely like what `Ctrl+B` does in VS Code to hide the sidebar). If I'm in the sidebar (e.g. the file tree), the same keystroke maximizes the sidebar.

To see how this works, simply install Eclipse. In that IDE you can hit `Ctrl+M` to maximize and minimize whatever has the focus, whether it's the editor, a view at the bottom, the project explorer on the side, etc. That's what we were wanting in #16927.

Importantly, I should hit the same keystroke to go back to _exactly_ the same "unmaximized" setting I was before—in other words, it remembers the size of the terminal, the editors, the sidebar, etc.

**Comment by dandavison** [comment]
> Importantly, I should hit the same keystroke to go back to exactly the same "unmaximized" setting I was before—in other words, it remembers the size of the terminal, the editors, the sidebar, etc.

I agree that remembering the previous editor configuration (splits) is important.

If this work involves changes that are generally in the direction of being able to save and restore previous editor configurations that would be fantastic. It's something [emacs does really well](https://www.gnu.org/software/emacs/manual/html_node/elisp/Window-Configurations.html).

**Comment by starball5** [comment]
Related question on Stack Overflow: [In VS Code, how can I toggle maximization of the current workbench part / editor group?](https://stackoverflow.com/q/58885067/11107541) where I have written up [a workaround](https://stackoverflow.com/a/77356431/11107541) using editor group maximization and zen mode. It supports Editor groups and Panels.

**Comment by lkraav** [comment]
Once you see how well IntelliJ does this out of the box, you can't unsee it.

**Comment by Chealer** [comment]
@benibenj wrote:
> Double clicking the first time:
>     […]
> 
> * Hides Side Panels & Auxiliary Bar

By “Side Panels & Auxiliary Bar”, do you mean sidebars, including the activity bar?
